### PR TITLE
Messages settings

### LINKED ISSUE #250: Expose default email config
Instead of dealing with updating default messages in a [file](https://github.com/DoSomething/gladiator/blob/master/app/Http/Utilities/Correspondence.php) this should be an admin config form in a settings menu.

It should update the defaults of any new messages created, but I think for ease of build, it shouldn't retroactivly update already created messages.

### PR DESCRIPTION
**Pull request by weerd**
#### What's this PR do?
This PR adds a new section for Settings and within that section there is a sub section for Default Messages Settings. It allows editing the default messages for when additional Contests are created.

#### How should this be manually tested?
`composer dump-autoload` reset the DB, re-migrate, and re-seed. Check that there's a new **messages_settings** table. Then login and go to Settings -> Messages and try editing the default messages. Next create a new Contest and see if the the edits to the defaults that are populated into the form are what you had previously changed.

#### Any background context you want to provide?
I attempted to create a generalized Settings system, with the intent that it could become a separate DS PHP package that we could pull into multiple projects. However, the complexity of Messages, with their multipart fields (subject, body, pro_tip, etc) made that approach too convoluted. Scrapped it but kept the code for a potential future implementation. 

While some of the code in this PR isn't ideal and could very well be refactored a bit, I wanted to get something up that works given I lost a bunch of time with the original approach.

#### What are the relevant tickets?
Fixes #250

---
@DoSomething/gladiator

### CODE REVIEW

**Comment by DFurnes** [comment]
@weerd Go back to bed!! 😷

File: app/Models/MessageSetting.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Database\Eloquent\Model;
+
+class MessageSetting extends Model
+{
+    /**
+     * Get the fillable attributes for the model.
+     *
+     * @return array
+     */
+    protected $fillable = ['type', 'key', 'label', 'subject', 'body', 'pro_tip', 'signoff', 'shoutout'];
```
Comment (angaither) [inline app/Models/MessageSetting.php]: @deadlybutter just a note here, you'll want to pull shoutout out of here when you make the pr for #280

File: database/migrations/2016_05_03_040602_create_messages_settings_table.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        Schema::create('messages_settings', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('type')->index();
+            $table->integer('key')->unsigned();
+            $table->string('label')->nullable();
+            $table->string('subject')->nullable();
+            $table->longText('body')->nullable();
+            $table->longtext('pro_tip')->nullable();
+            $table->longText('shoutout')->nullable();
```
Comment (angaither) [inline database/migrations/2016_05_03_040602_create_messages_settings_table.php]: here as well @deadlybutter

File: app/Models/MessageSetting.php (hunk 1 above)
Comment (deadlybutter) [inline app/Models/MessageSetting.php]: 👍

**Comment by angaither** [comment]
I tested and it works, going to merge this 🎉

**Pull request merged**